### PR TITLE
fixing register-env issue by using persift flag explicitly and adding…

### PR DESCRIPTION
**Pull request by anaggar**
… filtering capability

### CODE REVIEW

File: Tasks/WindowsMachineFileCopy/task.json [hunk 1]
```diff
@@ -110,7 +110,7 @@
             "defaultValue": "",
             "groupName": "advanced",
             "required": false,
-            "helpMarkDown": "This input is valid only for machine groups and is not supported for flat list of machines or output variables yet. Provide a list of machines like, dbserver.fabrikam.com, webserver.fabrikam.com, 192.168.12.34, or tags like, Role:DB; OS:Win8.1. If multiple tags are provided, then the task will run in all the machines with the specified tags. For Azure Resource Groups, provide the virtual machine's name like, ffweb, ffdb. The default is to run the task in all machines."
+            "helpMarkDown": "This input is valid only for machine groups and is not supported for flat list of machines or output variables yet. Provide a list of machines like, dbserver.fabrikam.com, webserver.fabrikam.com, 192.168.12.34, or tags like, Role:DB; OS:Win8.1. If multiple tags are provided, then the task will run in all the machines with the specified tags. The default is to run the task in all machines."
```
Comment (KrishnaAdityaB) [inline Tasks/WindowsMachineFileCopy/task.json]: This has to be corrected. Refer PSOnTargetMachines task

**Comment by KrishnaAdityaB** [comment]
:+1:

File: Tasks/WindowsMachineFileCopy/task.json (hunk 1 above)
Comment (anaggar) [inline Tasks/WindowsMachineFileCopy/task.json]: no it need not to be since we dont consume deploy rg flow in windows machine file copy

Comment (KrishnaAdityaB) [inline Tasks/WindowsMachineFileCopy/task.json]: Good point. Signed-off

**Pull request merged**